docs/flow.rst

```diff
@@ -0,0 +1,235 @@
+be executed, they will trigger events for the appropriate channels. These will
+be processed by the manager, which will carry out all the preparatory tasks
+necessary to start execution and then commuicate with a concrete workload
+management system so that the job can eventually be scheduled and run on a
+worker node.
+
+.. _Django Channels: https://github.com/django/channels
+
+Finally, the jobs are executed by the aptly named 'executors'. These are run on
+worker nodes and act as local execution managers: preparing a controlled
+execution environment, running the job's code, collecting results and
+communicating them back to the manager which stores them in the database.
+
+Utility Layers
+--------------
+
+Django's facilities are used for interfacing with the database, thus all models
+used in Flow are actually Django Model objects. The most important two models
+are the `Data` model and the `Process` model.
+
+A Data object represents a single instance of data to be processed, i.e. a node
+in the flow graph being executed.  It contains properties which mainly concern
+execution, such as various process and task IDs, output statuses and the results
+produced by executors.
+
+A Process object represents the way in which its Data object will be 'executed',
+i.e. the type of node in the flow graph and the associated code. It contains
+properties defining its relationship to other nodes in the graph currently being
+executed, the permissions that control access rights for users and other
+processes, and the actual code that is run by the executors.
+
+The code in the process object can be either final code that is already ready
+for execution, or it can be a form of template, for which an 'expression engine'
+is needed. An expression engine (the only one currently in use is `Jinja`_)
+pre-processes the process' code to produce text that can then be executed by an
+'execution engine'.
+
+.. _Jinja: http://jinja.pocoo.org
+
+An execution engine is, simply put, the interpreter that will run the processed
+code, just after an additional metadata discovery step. It is done by the
+execution engine because the encoding might be language-dependent. The
+properties to be discovered include process resource limits, secret
+requirements, etc. These properties are passed on to the executor, so that it
+can properly set up the target environment. The only currently supported
+execution engine is Bash.
+
+Technicalities
+==============
+
+The Manager
+-----------
+
+Being a Django Channels consumer application, the Flow Manager is entirely
+event-driven and mostly contextless. The main input events are data object
+creation, processing termination and intermediate executor messages. Once run,
+it consists of two distinct servers and a modularized connection framework used
+to interface with workload managers used by the deployment site.
+
+Dispatcher
+^^^^^^^^^^
+
+This is the central job scheduler. On receipt of an appropriate event through
```

Comment:
External job schedulers are outside the scope of this document and are consistently termed 'workload managers'. 'Central' is appropriate given it's conceptual position in the framework. It could scarcely be called internal, since it is nearly half of the entire space of services of this package (the other half being executors).